### PR TITLE
Disallow constructor arguments given multiple times.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by ekpyron — DISMISSED** [review]

**Review by axic — DISMISSED** [review]

**Comment by chriseth** [comment]
It seems this still has compilation failures in the end to end tests.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -2770,24 +2770,6 @@ BOOST_AUTO_TEST_CASE(virtual_function_usage_in_constructor_arguments)
 	ABI_CHECK(callContractFunction("getA()"), encodeArgs(2));
 }
 
-BOOST_AUTO_TEST_CASE(constructor_argument_overriding)
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:2773]: This was the test for the semantics of the feature we removed (overriding constructor arguments by giving them multiple times).

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -5505,10 +5487,10 @@ BOOST_AUTO_TEST_CASE(pass_dynamic_arguments_to_the_base_base)
 			uint public m_i;
 		}
 		contract Base1 is Base {
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:5489]: This test uses argument overwriting - I changed it to a test where arguments are passed on multiple times, I think this was more or less the intention.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -5528,9 +5510,11 @@ BOOST_AUTO_TEST_CASE(pass_dynamic_arguments_to_the_base_base_with_gap)
 			}
 			uint public m_i;
 		}
-		contract Base1 is Base(3) {}
+		contract Base1 is Base {
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:5513]: Here again, this test now tests that passing on arguments to a base that is two steps away in the linearized inheritance hierarchy, I also think this was the intention.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**